### PR TITLE
Check for correct peep state when removing mass

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This has exposed a bunch of things related to peep references that I want to explore. It looks like vehicles (that aren't ferriswheels) do not clean up after themselves when removing a peep from their vehicle. This should be changed as this will leave potentially invalid references in the vehicle.

### CODE REVIEW

File: src/openrct2/actions/SetCheatAction.cpp [hunk 1]
```diff
@@ -644,7 +644,11 @@ void SetCheatAction::RemoveAllGuests() const
                     auto peep = TryGetEntity<Guest>(peepInTrainIndex);
                     if (peep != nullptr)
                     {
-                        vehicle->mass -= peep->Mass;
+                        if ((peep->State == PeepState::OnRide && peep->RideSubState == PeepRideSubState::OnRide)
+                            || (peep->State == PeepState::LeavingRide && peep->RideSubState == PeepRideSubState::LeaveVehicle))
+                        {
+                            vehicle->mass -= peep->Mass;
```
Comment (IntelOrca) [inline src/openrct2/actions/SetCheatAction.cpp:650]:
```suggestion
if (vehicle->mass > peep->Mass)
    vehicle->mass -= peep->Mass;
else
    vehicle->mass = 0;
```

Comment (IntelOrca) [reply src/openrct2/actions/SetCheatAction.cpp:650]: Or
```suggestion
                            vehicle->ApplyMass(-peep->Mass);
```

Comment (duncanspumpkin) [reply src/openrct2/actions/SetCheatAction.cpp:650]: Why though? We don't verify in any other location adding/removing.

Comment (IntelOrca) [reply src/openrct2/actions/SetCheatAction.cpp:650]: Yeah we should do it though. Ideally we prevent all possible underflows / overflows. What reason is there not to?

Comment (duncanspumpkin) [reply src/openrct2/actions/SetCheatAction.cpp]: Okay I've added a function for it.

**Review by IntelOrca — APPROVED** [review]

**Pull request merged**